### PR TITLE
Revert frame timing changes

### PR DESCRIPTION
**Pull request by Archez**
(based on discussions in discord) It was found that the calculation changes in the `qpc_to_` methods were truncating/overflowing for a lot of values which lead to the vsync calculations to be incorrect and no longer drop frames as needed causing frame slowdown/slowmotion. On some machines this could manifest as early as 30 minutes of elapsed gameplay.

This reverts the changes to how they are in LUS 1.x which does not truncate/overflow.

This also adjusts the `do {} while` loop for `YieldProcessor` to a regular `while` loop so that we aren't yielding when we don't need to.

### CODE REVIEW

**Review by Malkierian — APPROVED** [review]

File: src/graphic/Fast3D/gfx_dxgi.cpp [hunk 1]
```diff
@@ -517,13 +517,11 @@ static void gfx_dxgi_handle_events(void) {
 }
 
 static uint64_t qpc_to_ns(uint64_t qpc) {
-    qpc *= 1000000000;
-    return qpc / dxgi.qpc_freq;
+    return qpc / dxgi.qpc_freq * 1000000000 + qpc % dxgi.qpc_freq * 1000000000 / dxgi.qpc_freq;
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_dxgi.cpp]: could be nice to make this a little less magic and throw
```cpp
#define MAGIC_NUMBER_NAME 1000000000
```
in there and replace this with

```suggestion
    return qpc / dxgi.qpc_freq * MAGIC_NUMBER_NAME + qpc % dxgi.qpc_freq * MAGIC_NUMBER_NAME / dxgi.qpc_freq;
```

File: src/graphic/Fast3D/gfx_dxgi.cpp [hunk 2]
```diff
@@ -517,13 +517,11 @@ static void gfx_dxgi_handle_events(void) {
...
 }
 
 static uint64_t qpc_to_100ns(uint64_t qpc) {
-    qpc *= 10000000;
-    return qpc / dxgi.qpc_freq;
+    return qpc / dxgi.qpc_freq * 10000000 + qpc % dxgi.qpc_freq * 10000000 / dxgi.qpc_freq;
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_dxgi.cpp]: same comment as above
```cpp
#define MAGIC_NUMBER_NAME 10000000
```

```suggestion
    return qpc / dxgi.qpc_freq * MAGIC_NUMBER_NAME + qpc % dxgi.qpc_freq * MAGIC_NUMBER_NAME / dxgi.qpc_freq;
```

**Review by briaguya0 — COMMENTED** [review]
non blocking but left some suggestions to move away from magic numbers

i'm not sure what the best names for what i have as `MAGIC_NUMBER_NAME` are, my initial thought was `QPC_PER_NS` and `QPC_PER_100NS` but that feels wrong considering the `qpc_to_ns` magic number is 100x *bigger* than the `qpc_to_100ns` magic number

File: src/graphic/Fast3D/gfx_dxgi.cpp (hunk 1 above)
Comment (Spodi) [reply src/graphic/Fast3D/gfx_dxgi.cpp]: Your magic number is just the factor from seconds to ns. Or seconds to 100ns steps respectively. It isn't really directly related to the performance counter. ( qpc [ticks] / qpc_freq [ticks/sec] = time [sec] )

Comment (Archez) [reply src/graphic/Fast3D/gfx_dxgi.cpp]: I've updated the PR to add defines for these constants

**Review by briaguya0 — APPROVED** [review]

**Pull request merged**